cardinal/system/t_meat_puppet_system_test.go

```diff
@@ -0,0 +1,290 @@
+package system
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/component"
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/constants"
+	"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/enums"
+	"github.com/stretchr/testify/assert"
+)
+
+func setup() {
+	ts = NewTokeniserSystem()
+}
+
+// TestProcessCommandsTokensLogic tests the ProcessCommandsTokensLogic function
+func TestProcessCommandsTokensLogic(t *testing.T) {
+	tests := []struct {
+		name         string
+		tokens       []string
+		player       component.Player
+		expectedOut  string
+		expectedMove bool
+		expectedErr  uint8
+	}{
+		{
+			name:         "Too many tokens",
+			tokens:       []string{"THROW", "THE", "BALL", "TO", "THE", "WINDOW", "LOCK", "THE", "DOOR", "TO", "THE", "PATH", "BURN", "THE", "BOTTLE", "AT", "THE", "STAIRS"},
+			player:       component.Player{PlayerID: 6, RoomID: 0},
+			expectedOut:  constants.ErrParserRoutineTKCX.Message,
+			expectedMove: false,
+			expectedErr:  constants.ErrParserRoutineTKCX.Code,
+		},
+		{
+			name:         "Go command",
+			tokens:       []string{"GO", "TO", "PATH"},
+			player:       component.Player{PlayerID: 6, RoomID: 0},
+			expectedOut:  "GOING TO NEXT ROOM - DIRECTION SYSTEM - TO BE IMPLEMENTED",
+			expectedMove: true,
+			expectedErr:  0,
+		},
+		{
+			name:         "Invalid command",
+			tokens:       []string{"SEE", "UP"},
+			player:       component.Player{PlayerID: 6, RoomID: 0},
+			expectedOut:  constants.ErrParserRoutineNOP.Message,
+			expectedMove: false,
+			expectedErr:  constants.ErrParserRoutineNOP.Code,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			output, move, err := ProcessCommandsTokensLogicTest(tt.tokens, tt.player.PlayerID, tt.player.RoomID)
+
+			assert.Equal(t, tt.expectedOut, output)
+			assert.Equal(t, tt.expectedMove, move)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleAlias tests the handleAlias function
+func TestHandleAlias(t *testing.T) {
+	setup()
+	tests := []struct {
+		name        string
+		tokens      []string
+		playerID    uint32
+		expectedErr uint8
+	}{
+		{
+			name:        "Inventory alias",
+			tokens:      []string{"INVENTORY"},
+			playerID:    6,
+			expectedErr: 135,
+		},
+		{
+			name:        "Look alias",
+			tokens:      []string{"LOOK", "UP"},
+			playerID:    6,
+			expectedErr: 130,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := handleAliasTest(tt.tokens, tt.playerID)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleVerb tests the handleVerb function
+func TestHandleVerb(t *testing.T) {
```

Comment:
as above split into 2 tests

Reply:
Leaving this file until get response from Argus guys